src/bayesian/log_posterior.py

```diff
@@ -43,12 +43,12 @@ def initialize_pool_variables(local_min, local_max, local_emulation_config, loca
     - Store in global variable for efficient reuse during MCMC
     """
 
-    global g_min  # noqa: PLW0603
-    global g_max  # noqa: PLW0603
-    global g_emulation_config  # noqa: PLW0603
-    global g_emulation_results  # noqa: PLW0603
-    global g_experimental_results  # noqa: PLW0603
-    global g_emulator_cov_unexplained  # noqa: PLW0603
```

Comment:
Please don't drop the linting statements. We have to deal with the global state here (otherwise I found performance was poor), but I'm aiming to keep this codebase quality high, so linting tools like these can help